src/public/modules/forms/admin/directives/edit-form.client.directive.js

```diff
@@ -98,13 +98,10 @@ function editFormController(
   let hiddenFieldSet = new Set() // To monitor which fields should be hidden because of form logic
   let conditionFieldSet = new Set() // To monitor which fields are conditions in the form logic
   // On change of form logic (scope.myform.form_logics), update our logic monitors
-  $scope.$watch(
-    (scope) => scope.myform.form_logics,
-    function (_newVal, _oldVal) {
-      updateHiddenFieldSet()
-      updateConditionFieldSet()
-    },
-  )
+  $scope.$watchCollection('myform.form_logics', function (_newVal, _oldVal) {
```

Comment:
nit: since both values are unused, i think we can just call

```ts
$scope.$watchCollection('myform.form_logics', function () {
  ...
}
```

However, this is also indicative of a code smell, but probably no need to fix it since React is coming

Reply:
Will leave this for now~